### PR TITLE
fix: Rebuilding the index only latest historical entry should be indexed

### LINKED ISSUE #413: Rebuilding the index with 10 or more historical entries does not index 'current' entry
**Describe the bug**
When rebuilding the index and you have 10 or more historical entries for a particular resource. We would not index the latest resource.

This occurs since we fetch all resources including the historical entries.

**To Reproduce**
1. Update a resource 10 or more times
2. Rebuild the index
3. Use MongoDB Compass to verify that our index is no longer pointing to the latest resource, but will instead point to i.e: `/{typename}/_history/9`

**Expected behavior**
We should only index the `current` active resource and not the historical ones.

### PR DESCRIPTION
**Pull request by whyfate**
When rebuilding the index we should only index the latest historical 
entry per each resource.

### CODE REVIEW

**Comment by kennethmyhra** [comment]
Hey @whyfate, could you show me the query string for the search you are trying to execute

**Comment by whyfate** [comment]
@kennethmyhra  It's not a query operation.
I upgrade the `Spark`,so i need rebuild the index to slove some question.
When `IndexRebuildService.cs` rebuilds the index, the data sorting of the query has the above problems.
`IndexRebuildService.cs`  will rebuild the index of `_history/11` first and rebuild the index of `_history/9` later.
The resource should be deleted, and the index will be deleted. Now it is not deleted,`_history/9` has saved it.

**Comment by kennethmyhra** [comment]
I looked through the code, and this should be changed in a slightly different way:
The line of code here: https://github.com/FirelyTeam/spark/blob/3152e9dabe510ddb4198cf56d777e2e3532f36e2/src/Spark.Mongo/Store/MongoFhirStorePagedReader.cs#L24

Should be changed to:
```csharp
var filter = Builders<BsonDocument>.Filter.Eq(Field.STATE, Value.CURRENT);
```

Do you want to apply that change and resubmit the PR?

**Comment by whyfate** [comment]
Yes, you're right. We just need to rebuild the index of  the final data, and we don't need to repeat all the operations.
I'll do it.

File: src/Spark.Mongo/Store/MongoCollectionPageResult.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public async Task IterateAllPagesAsync(Func<IReadOnlyList<T>, Task> callback)
             for (var offset = 0; offset < TotalRecords; offset += _pageSize)
             {
                 var data = await _collection.Find(_filter)
-                    .Sort(Builders<BsonDocument>.Sort.Ascending(Field.PRIMARYKEY))
+                    .Sort(Builders<BsonDocument>.Sort.Ascending(Field.REFERENCE).Ascending(Field.WHEN))
```
Comment (kennethmyhra) [inline src/Spark.Mongo/Store/MongoCollectionPageResult.cs]: This part needs to be reverted

**Review by kennethmyhra — CHANGES_REQUESTED** [review]

File: src/Spark.Mongo/Store/MongoCollectionPageResult.cs (hunk 1 above)
Comment (whyfate) [reply src/Spark.Mongo/Store/MongoCollectionPageResult.cs]: Wait a minute, I'm not familiar with the operation of GitHub.

**Review by kennethmyhra — APPROVED** [review]

**Pull request merged**